spring-cloud-aws-sns/src/main/java/io/awspring/cloud/sns/core/SnsAsyncTopicArnResolver.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2013-2025 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.awspring.cloud.sns.core;
+
+import org.springframework.util.Assert;
+import software.amazon.awssdk.arns.Arn;
+import software.amazon.awssdk.services.sns.SnsAsyncClient;
+
+/**
+ * A {@link TopicArnResolver} implementation to determine topic ARN by name against an {@link SnsAsyncClient}.
+ *
+ * @author Artem Bilan
+ *
+ * @since 4.0
+ */
+public class SnsAsyncTopicArnResolver implements TopicArnResolver {
+	private final SnsAsyncClient snsClient;
```

Comment:
Currently we don't do autoconfiguration Async client for SNS. Mb something to think of.

I am thinking mb we can switch this integration to use:
`DefaultTopicArnResolver` since it supports fifo queues creation.

```
	// fix for https://github.com/awspring/spring-cloud-aws/issues/707
			if (topicName.endsWith(".fifo")) {
				builder.attributes(Map.of("FifoTopic", "true"));
			}